src/foundry/prosemirror/schema/image-link-node.d.mts

```diff
@@ -0,0 +1,29 @@
+import { MarkSpec, Node, NodeSpec } from "prosemirror-model";
+import type SchemaDefinition from "./schema-definition.d.mts";
+import { EditorView } from "prosemirror-view";
+
+export default ImageLinkNode;
+
+/**
+ * A class responsible for encapsulating logic around image-link nodes in the ProseMirror schema.
+ */
+declare class ImageLinkNode extends SchemaDefinition {
+  static override tag: "a";
+
+  static override get attrs(): Record<string, any>;
+
+  static override getAttrs(el: HTMLElement): boolean | Record<string, any>;
```

Comment:
Perhaps this should be something more specific like `HTMLLinkElement`. If this applies here it probably applies everywhere.

Also this might mean that it's `HTMLLinkElement`'s specific attributes but I don't think we have to get that specific.